File: test2/src/test/java/TestCtrip.java
import org.apache.commons.io.FileUtils;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.*;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.PageFactory;

import java.io.File;
import java.text.SimpleDateFormat;
import java.util.Date;

public class TestCtrip {
    private WebDriver driver;

    @BeforeEach
    public void setup() {
        // Windows：以下路径改为你本地到webdriver路径
        String driver_path = "C:/Data/repos/software_test/test1/driver/chromedriver.exe";
        System.setProperty("webdriver.chrome.driver", driver_path);
        ChromeOptions chromeOptions = new ChromeOptions();
        chromeOptions.addArguments("--remote-allow-origins=*");
        driver = new ChromeDriver(chromeOptions);
        driver.get("https://www.ctrip.com");
        driver.manage().window().maximize();
    }

    @Test
    public void test_Ctrip_R005() throws InterruptedException {
        int cnt = 0;
        String[] departure_cities = {"济南", "天津"};
        String[] arrival_cities = {"西安", "杭州"};

        PageFactory.initElements(driver, this);

        for (String departure_city : departure_cities) {
            for (String arrival_city : arrival_cities) {
                driver.findElement(By.xpath("//*[@id=\"leftSideNavLayer\"]/div/div/div[1]/div/div")).click();
                Actions actions = new Actions(driver);
                WebElement webElement = driver.findElement(By.xpath("//*[@id=\"leftSideNavLayer\"]/div/div/div[2]/div/div[1]/div/div[3]/button"));
                actions.moveToElement(webElement).click().perform();

                driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[1]/div/ul/li[2]")).click();

                driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).click();
                driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).clear();
                driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).sendKeys(departure_city);
                Thread.sleep(1000);

                driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).click();
                driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).clear();
                driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).sendKeys(arrival_city);
                Thread.sleep(1000);

                driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]")).click();
                driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]")).click();

                Actions actionsTwo = new Actions(driver);
                actionsTwo.moveToElement(driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]"))).click().perform();

                if(cnt == 0)
                    driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[1]/div/div[2]/div[2]/ul[2]/li[5]")).click();

                driver.findElement(By.xpath("//button[@class=\"btn-blue btn-search\"]")).click();
                Thread.sleep(1000);

                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/div")).click();
                Thread.sleep(1000);

                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[1]/div[1]")).click();
                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[2]")).click();
                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[3]/div[2]")).click();
                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[5]/div[1]")).click();
                if(cnt != 3)
                    driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[6]/div[1]")).click();

                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/div")).click();
                Thread.sleep(1000);

                takeScreenshot("test_Ctrip_R005_00" + Integer.toString(cnt + 1) + ".png");
                cnt++;
            }
        }
    }

        @Test
    public void test_Ctrip_R006() throws InterruptedException {
        int cnt = 0;
        String[] departure_cities = {"兰州", "徐州"};
        String[] arrival_cities = {"郑州", "厦门"};

        PageFactory.initElements(driver, this);

        for (String departure_city : departure_cities) {
            for (String arrival_city : arrival_cities) {
                driver.findElement(By.xpath("//*[@id=\"leftSideNavLayer\"]/div/div/div[1]/div/div")).click();
                Actions actions = new Actions(driver);
                WebElement webElement = driver.findElement(By.xpath("//*[@id=\"leftSideNavLayer\"]/div/div/div[2]/div/div[1]/div/div[3]/button"));
                actions.moveToElement(webElement).click().perform();

                driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[1]/div/ul/li[2]")).click();

                driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).click();
                driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).clear();
                driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).sendKeys(departure_city);
                Thread.sleep(1000);

                driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).click();
                driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).clear();
                driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).sendKeys(arrival_city);
                Thread.sleep(1000);

                driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]")).click();
                driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]")).click();

                Actions actionsTwo = new Actions(driver);
                actionsTwo.moveToElement(driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]"))).click().perform();

                if(cnt == 0)
                    driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[1]/div/div[2]/div[2]/ul[2]/li[5]")).click();

                driver.findElement(By.xpath("//button[@class=\"btn-blue btn-search\"]")).click();
                Thread.sleep(1000);

                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/div")).click();
                Thread.sleep(1000);

                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[1]/div[1]")).click();
                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[2]/div[2]")).click();
                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[3]/div[2]")).click();
                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[5]/div[1]")).click();
                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[6]/div[1]")).click();

                driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/div")).click();
                Thread.sleep(1000);

                takeScreenshot("test_Ctrip_R006_00" + Integer.toString(cnt + 1) + ".png");
                cnt++;
            }
        }
    }

    @Test
    public void test_Ctrip_R007() throws InterruptedException {
        PageFactory.initElements(driver, this);
        driver.findElement(By.xpath("//*[@id=\"leftSideNavLayer\"]/div/div/div[1]/div/div")).click();
        Actions actions = new Actions(driver);
        WebElement webElement = driver.findElement(By.xpath("//*[@id=\"leftSideNavLayer\"]/div/div/div[2]/div/div[1]/div/div[3]/button/span[2]"));
        actions.moveToElement(webElement).click().perform();

        driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[1]/div/ul/li[2]")).click();

        driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).click();
        driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).clear();
        driver.findElement(By.xpath("//*[@id=\"label-departStation\"]")).sendKeys("123456");
        Thread.sleep(1000);

        driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).click();
        driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).clear();
        driver.findElement(By.xpath("//input[@placeholder=\"到达城市\"]")).sendKeys("海口");
        Thread.sleep(1000);

        driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[1]/div/div[2]/div[2]/ul[2]/li[5]")).click();

        driver.findElement(By.xpath("//button[@class=\"btn-blue btn-search\"]")).click();
        Thread.sleep(1000);

        takeScreenshot("test_Ctrip_R007_001.png");
    }

    @Test
    public void test_Ctrip_R008() throws InterruptedException {
        PageFactory.initElements(driver, this);
        driver.findElement(By.xpath("//*[@id=\"leftSideNavLayer\"]/div/div/div[1]/div/div")).click();
        Actions actions = new Actions(driver);
        WebElement webElement = driver.findElement(By.xpath("//*[@id=\"leftSideNavLayer\"]/div/div/div[2]/div/div[1]/div/div[3]/button/span[2]"));
        actions.moveToElement(webElement).click().perform();

        driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[1]/div/ul/li[2]")).click();
        driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[3]")).click();
        Thread.sleep(1000);

        driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"app\"]/div[2]/div[1]/div[2]/div[1]/div/div[2]/div[1]/ul[2]/li[31]")).click();

        driver.findElement(By.xpath("//button[@class=\"btn-blue btn-search\"]")).click();
        Thread.sleep(1000);

        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/ul[1]/li[3]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/div")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[2]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[3]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[4]")).click();
        Thread.sleep(1000);

        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/div")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[3]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[4]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[5]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[6]/div[1]")).click();
        Thread.sleep(1000);

        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[4]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[5]/div[1]")).click();
        driver.findElement(By.xpath("//*[@id=\"__next\"]/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[6]/div[1]")).click();
        Thread.sleep(1000);

        takeScreenshot("test_Ctrip_R008_001.png");
    }

    @AfterEach
    public void teardown() {
        this.driver.quit();
    }

    private void takeScreenshot(String fileName) {
        SimpleDateFormat dateFormat = new SimpleDateFormat("HHmmssddSSS");
        String timestamp = dateFormat.format(new Date());
        String timestampedFileName = timestamp + "_" + fileName;
        File screenshotsDir = new File("screenshots/java");
        if (!screenshotsDir.exists()) {
            screenshotsDir.mkdirs();
        }
        String screenshotFilePath = screenshotsDir.getAbsolutePath() + File.separator + timestampedFileName;
        File screenshotFile = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
        try {
            FileUtils.copyFile(screenshotFile, new File(screenshotFilePath));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: test2/src/test/python/TestCtrip.py
import os
import time
import pytest
from datetime import datetime
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.action_chains import ActionChains


@pytest.fixture( scope="function" )
def init_browser():
    service = Service(executable_path="C:/Data/repos/software_test/test1/driver/chromedriver.exe")
    driver = webdriver.Chrome(service=service)
    driver.get("https://www.ctrip.com/")
    driver.maximize_window()
    yield driver
    driver.quit()


def take_screenshot(driver, file_name):
    timestamp = datetime.now().strftime( "%H%M%S%d%f" )
    timestamped_file_name = f"{timestamp}_{file_name}"
    screenshots_dir = "screenshots/python"
    if not os.path.exists( screenshots_dir ):
        os.makedirs( screenshots_dir )
    screenshot_file_path = os.path.join( screenshots_dir, timestamped_file_name )
    driver.save_screenshot( screenshot_file_path )


# ----------------- test_Ctrip_R005 -----------------
def test_Ctrip_R005(init_browser):
    driver = init_browser
    cnt = 0
    departure_cities = ["济南", "天津"]
    arrival_cities = ["西安", "杭州"]

    for departure_city in departure_cities:
        for arrival_city in arrival_cities:
            driver.find_element(By.XPATH, "//*[@id='leftSideNavLayer']/div/div/div[1]/div/div").click()
            actions = ActionChains(driver)
            web_element = driver.find_element(By.XPATH,
                "//*[@id='leftSideNavLayer']/div/div/div[2]/div/div[1]/div/div[3]/button")
            actions.move_to_element(web_element).click().perform()

            driver.find_element(By.XPATH, "//*[@id='app']/div[2]/div[1]/div[1]/div/ul/li[2]").click()

            depart_input = driver.find_element(By.XPATH, "//*[@id='label-departStation']")
            depart_input.click()
            depart_input.clear()
            depart_input.send_keys(departure_city)
            time.sleep(1)

            arrive_input = driver.find_element(By.XPATH, "//input[@placeholder='到达城市']")
            arrive_input.click()
            arrive_input.clear()
            arrive_input.send_keys(arrival_city)
            time.sleep(1)

            driver.find_element(By.XPATH, "//*[@id='app']/div[2]/div[1]").click()
            date_input = driver.find_element(By.XPATH,
                "//*[@id='app']/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]")
            date_input.click()

            actions_two = ActionChains(driver)
            actions_two.move_to_element(date_input).click().perform()

            if cnt == 0:
                driver.find_element(By.XPATH,
                    "//*[@id='app']/div[2]/div[1]/div[2]/div[1]/div/div[2]/div[2]/ul[2]/li[5]").click()

            driver.find_element(By.XPATH, "//button[@class='btn-blue btn-search']").click()
            time.sleep(1)

            driver.find_element(By.XPATH,
                "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/div").click()
            time.sleep(1)

            click_paths = [
                "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[1]/div[1]",
                "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[2]",
                "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[3]/div[2]",
                "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[5]/div[1]",
            ]
            for path in click_paths:
                driver.find_element(By.XPATH, path).click()
            if cnt != 3:
                driver.find_element(By.XPATH,
                    "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[6]/div[1]").click()

            driver.find_element(By.XPATH,
                "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/div").click()
            time.sleep(1)

            take_screenshot(driver, f"test_Ctrip_R005_00{cnt + 1}.png")
            cnt += 1


# ----------------- test_Ctrip_R006 -----------------
def test_Ctrip_R006(init_browser):
    driver = init_browser
    cnt = 0
    departure_cities = ["兰州", "徐州"]
    arrival_cities = ["郑州", "厦门"]

    for departure_city in departure_cities:
        for arrival_city in arrival_cities:
            driver.find_element(By.XPATH, "//*[@id='leftSideNavLayer']/div/div/div[1]/div/div").click()
            actions = ActionChains(driver)
            web_element = driver.find_element(By.XPATH,
                "//*[@id='leftSideNavLayer']/div/div/div[2]/div/div[1]/div/div[3]/button")
            actions.move_to_element(web_element).click().perform()

            driver.find_element(By.XPATH, "//*[@id='app']/div[2]/div[1]/div[1]/div/ul/li[2]").click()

            depart_input = driver.find_element(By.XPATH, "//*[@id='label-departStation']")
            depart_input.click()
            depart_input.clear()
            depart_input.send_keys(departure_city)
            time.sleep(1)

            arrive_input = driver.find_element(By.XPATH, "//input[@placeholder='到达城市']")
            arrive_input.click()
            arrive_input.clear()
            arrive_input.send_keys(arrival_city)
            time.sleep(1)

            driver.find_element(By.XPATH, "//*[@id='app']/div[2]/div[1]").click()
            date_input = driver.find_element(By.XPATH,
                "//*[@id='app']/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]")
            date_input.click()

            actions_two = ActionChains(driver)
            actions_two.move_to_element(date_input).click().perform()

            if cnt == 0:
                driver.find_element(By.XPATH,
                    "//*[@id='app']/div[2]/div[1]/div[2]/div[1]/div/div[2]/div[2]/ul[2]/li[5]").click()

            driver.find_element(By.XPATH, "//button[@class='btn-blue btn-search']").click()
            time.sleep(1)

            driver.find_element(By.XPATH,
                "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/div").click()
            time.sleep(1)

            click_paths = [
                "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[1]/div[1]",
                "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[2]/div[2]",
                "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[3]/div[2]",
                "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[5]/div[1]",
                "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[6]/div[1]",
            ]
            for path in click_paths:
                driver.find_element(By.XPATH, path).click()

            driver.find_element(By.XPATH,
                "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/div").click()
            time.sleep(1)

            take_screenshot(driver, f"test_Ctrip_R006_00{cnt + 1}.png")
            cnt += 1


# ----------------- test_Ctrip_R007 -----------------
def test_Ctrip_R007(init_browser):
    driver = init_browser

    driver.find_element(By.XPATH, "//*[@id='leftSideNavLayer']/div/div/div[1]/div/div").click()
    actions = ActionChains(driver)
    web_element = driver.find_element(By.XPATH,
        "//*[@id='leftSideNavLayer']/div/div/div[2]/div/div[1]/div/div[3]/button/span[2]")
    actions.move_to_element(web_element).click().perform()

    driver.find_element(By.XPATH, "//*[@id='app']/div[2]/div[1]/div[1]/div/ul/li[2]").click()

    depart_input = driver.find_element(By.XPATH, "//*[@id='label-departStation']")
    depart_input.click()
    depart_input.clear()
    depart_input.send_keys("123456")
    time.sleep(1)

    arrive_input = driver.find_element(By.XPATH, "//input[@placeholder='到达城市']")
    arrive_input.click()
    arrive_input.clear()
    arrive_input.send_keys("海口")
    time.sleep(1)

    driver.find_element(By.XPATH, "//*[@id='app']/div[2]/div[1]").click()
    driver.find_element(By.XPATH,
        "//*[@id='app']/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]").click()
    driver.find_element(By.XPATH,
        "//*[@id='app']/div[2]/div[1]/div[2]/div[1]/div/div[2]/div[2]/ul[2]/li[5]").click()

    driver.find_element(By.XPATH, "//button[@class='btn-blue btn-search']").click()
    time.sleep(1)

    take_screenshot(driver, "test_Ctrip_R007_001.png")


# ----------------- test_Ctrip_R008 -----------------
def test_Ctrip_R008(init_browser):
    driver = init_browser

    driver.find_element(By.XPATH, "//*[@id='leftSideNavLayer']/div/div/div[1]/div/div").click()
    actions = ActionChains(driver)
    web_element = driver.find_element(By.XPATH,
        "//*[@id='leftSideNavLayer']/div/div/div[2]/div/div[1]/div/div[3]/button/span[2]")
    actions.move_to_element(web_element).click().perform()

    driver.find_element(By.XPATH, "//*[@id='app']/div[2]/div[1]/div[1]/div/ul/li[2]").click()
    driver.find_element(By.XPATH, "//*[@id='app']/div[2]/div[1]/div[3]").click()
    time.sleep(1)

    driver.find_element(By.XPATH,
        "//*[@id='app']/div[2]/div[1]/div[2]/div[4]/div[1]/div[1]").click()
    driver.find_element(By.XPATH,
        "//*[@id='app']/div[2]/div[1]/div[2]/div[1]/div/div[2]/div[1]/ul[2]/li[31]").click()

    driver.find_element(By.XPATH, "//button[@class='btn-blue btn-search']").click()
    time.sleep(1)

    driver.find_element(By.XPATH, "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/ul[1]/li[3]").click()
    driver.find_element(By.XPATH, "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/div").click()
    click_paths = [
        "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[1]",
        "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[2]",
        "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[3]",
        "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[2]/div[4]",
    ]
    for path in click_paths:
        driver.find_element(By.XPATH, path).click()
    time.sleep(1)

    driver.find_element(By.XPATH, "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/div").click()
    click_paths2 = [
        "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[3]/div[1]",
        "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[4]/div[1]",
        "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[5]/div[1]",
        "//*[@id='__next']/div/div[3]/div/div[1]/div[2]/div[3]/ul/li[6]/div[1]",
    ]
    for path in click_paths2:
        driver.find_element(By.XPATH, path).click()
    time.sleep(1)

    click_paths3 = [
        "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[4]/div[1]",
        "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[5]/div[1]",
        "//*[@id='__next']/div/div[3]/div/div[3]/div[2]/div[3]/ul/li[6]/div[1]",
    ]
    for path in click_paths3:
        driver.find_element(By.XPATH, path).click()
    time.sleep(1)

    take_screenshot(driver, "test_Ctrip_R008_001.png")
